domain/status/modelmigration/import.go

```diff
@@ -32,12 +68,66 @@ func (i *importOperation) Name() string {
 // Setup the import operation.
 // This will create a new service instance.
 func (i *importOperation) Setup(scope modelmigration.Scope) error {
+	i.service = service.NewService(
+		state.NewState(scope.ModelDB(), i.clock, i.logger),
+		i.clock,
+		i.logger,
+		// TODO(jack): This is currently the wrong logger. We should construct
+		// the StatusHistory using the model logger, however, at the moment, we
+		// cannot get the model logger until the model has been imported. Once
+		// this has changed, refactor this to use the model logger.
+		domain.NewStatusHistory(i.logger, i.clock),
+	)
 	return nil
 }
 
 // Execute the import, loading the statuses of the various entities out of the
 // description representation, into the domain.
 func (i *importOperation) Execute(ctx context.Context, model description.Model) error {
-	// TODO: Import statuses here, instead of the application domain
+	for _, app := range model.Applications() {
+		appStatus := i.importStatus(app.Status())
+		if err := i.service.SetApplicationStatus(ctx, app.Name(), &appStatus); err != nil {
+			return err
+		}
+
+		for _, unit := range app.Units() {
+			unitName, err := coreunit.NewName(unit.Name())
+			if err != nil {
+				return err
+			}
+			unitAgentStatus := i.importStatus(unit.AgentStatus())
+			if err := i.service.SetUnitAgentStatus(ctx, unitName, &unitAgentStatus); err != nil {
+				return err
+			}
+
+			unitWorkloadStatus := i.importStatus(unit.WorkloadStatus())
+			if err := i.service.SetUnitWorkloadStatus(ctx, unitName, &unitWorkloadStatus); err != nil {
+				return err
+			}
```

Comment:
I think we probably want these to be 1 call in the end, short term this is fine.